app/controllers/api/v1/registrations/registrations_controller.rb

```diff
@@ -54,8 +55,24 @@ def create
     render json: { status: 'bad request', message: 'You need to supply at least one lane' }, status: :bad_request
   end
 
+  def user_can_create_registration
+    @target_user = User.find(params.require('user_id'))
+    @competition = Competition.find(params.require('competition_id'))
+    raise WcaExceptions::RegistrationError.new(:forbidden, Registrations::ErrorCodes::REGISTRATION_ALREADY_EXISTS) if
+      Registration.exists?(competition: @competition, user: @target_user)
+
+    # Only the user themselves can create a registration for the user
+    raise WcaExceptions::RegistrationError.new(:unauthorized, Registrations::ErrorCodes::USER_INSUFFICIENT_PERMISSIONS) unless @current_user.id == @target_user.id
+
+    # Only organizers can register when registration is closed, and they can only register for themselves - not for other users
+    raise WcaExceptions::RegistrationError.new(:forbidden, Registrations::ErrorCodes::REGISTRATION_CLOSED) unless @competition.registration_currently_open? || organizer_modifying_own_registration?(@competition, @current_user, @target_user)
```

Comment:
Not sure how nitpicky we want to get here, but at this point target_user == current_user has already been established (otherwise the previous exception would've been thrown). So checking `can_manage_competition` should be enough.

Reply:
I am aware that this would make validation checks order-dependent, so this is not a strong preference or requirement to change it. More of an FYI in case you have any thoughts about it.

Reply:
these are already order dependent so we do not leak ban information. E.g checking if you are the target user before checking you are banned. so this is fine? I'll check if the tests still pass